Add OSM tutorial and docs

johnsonshi opened the pull request:
Add OSM tutorial and docs.

Reference PR that implemented support for
OSM progressive traffic shifting in Flagger: #955.

Signed-off-by: Johnson Shi <Johnson.Shi@microsoft.com>

stefanprodan commented on docs/gitbook/tutorials/osm-progressive-delivery.md:
```diff
@@ -0,0 +1,360 @@
+kubectl patch \
+deployment flagger-loadtester \
+-n test \
+-p '{"spec": {"template": {"metadata": {"annotations": {"openservicemesh.io/inbound-port-exclusion-list": "80, 8080"}}}}}' \
```
You’ve added this annotation in the previous PR. I think we should remove this instruction and the restart one from below as it does nothing.

johnsonshi replied:
Removed

stefanprodan commented on docs/gitbook/tutorials/osm-progressive-delivery.md:
````diff
@@ -0,0 +1,360 @@
+Install Flagger in the `osm-system` namespace.
+
+```bash
+kubectl apply -k github.com/fluxcd/flagger//kustomize/osm
````
```suggestion
kubectl apply -k https://github.com/fluxcd/flagger//kustomize/osm?ref=main
```

I would also add here the Helm install command for those that prefer Helm.

johnsonshi replied:
Added helm installation as an alternative.

stefanprodan commented on docs/gitbook/tutorials/osm-progressive-delivery.md:
```diff
@@ -0,0 +1,360 @@
+  name: podinfo
+  namespace: test
+spec:
+  # deployment reference
```
Let’s add here `provider: osm`, even if it’s not required, it allows people to mix ingress controllers with OSM.

johnsonshi replied:
Added

johnsonshi commented:
Hi @stefanprodan (cc @draychev), I've addressed all PR comments and this can now be merged. Additionally, the Flagger controller needs to have a new release version cut so that the latest changes (including Flagger OSM support) is pushed to the Flagger controller container registries. Thank you!

stefanprodan reviewed (approved):
LGTM

Thanks @johnsonshi
